jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -197,6 +198,58 @@ def __init__(self, *args, **kwargs):
         # default_tool_priority=['jdaviz:selectslice']
         super().__init__(*args, **kwargs)
 
+    def _get_spatial_subset_layers(self):
+        return [ls for ls in self.state.layers
+                if isinstance(getattr(ls.layer, 'subset_state', None), RoiSubsetState)]
+
+    def _get_spectral_subset_layers(self):
+        return [ls for ls in self.state.layers
+                if isinstance(getattr(ls.layer, 'subset_state', None), RangeSubsetState)]
+
+    def _get_marks_for_layers(self, layers):
+        layers_list = list(self.state.layers)
+        # here we'll assume that all custom marks are subclasses of Lines/GL but don't directly
+        # use Lines/LinesGL (so an isinstance check won't be sufficient here)
+        layer_marks = [m for m in self.figure.marks if m.__class__.__name__ in ['Lines', 'LinesGL']]
```

Comment:
Didn't @duytnguyendtn write basically the same loop just recently somewhere else? Looks so familiar.

Reply:
hmmm, good point.  If that's already in main, I could try to replace both of them with a `viewer.custom_marks` property or something 🤔

Reply:
Maybe I am thinking about https://github.com/spacetelescope/jdaviz/pull/1550 that has been merged.

Reply:
Not sure that I have a strong opinion here. Though I guess my immediate expectation would be more of a nested structure, like `viewer.marks` containing the native marks and then you could get the custom via `viewer.marks.custom`. But my opinion isn't strong